src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -301,12 +331,362 @@ TEST_F(RPCServiceImplTest,
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  typedef am::plugin_manager::RPCPlugin RPCPlugin;
+  utils::Optional<RPCPlugin> mock_rpc_plugin_opt(
+      utils::Optional<RPCPlugin>::OptionalEmpty::EMPTY);
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(kFunctionId, source))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+
+  EXPECT_FALSE(rpc_service_->IsAppServiceRPC(function_id, source));
+}
+
+TEST_F(RPCServiceImplTest, IsAppServiceRPC_PluginWithWrongName_ReturnFalse) {
+  int32_t function_id = kFunctionId;
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  utils::Optional<am::plugin_manager::RPCPlugin> mock_rpc_plugin_opt(
+      mock_rpc_plugin_);
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(_, _))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+  ON_CALL(mock_rpc_plugin_, PluginName()).WillByDefault(Return(""));
+  EXPECT_FALSE(rpc_service_->IsAppServiceRPC(function_id, source));
 }
 
+TEST_F(RPCServiceImplTest,
+       IsAppServiceRPC_CommandFactoryIsAbleToProcess_ReturnTrue) {
+  int32_t function_id = kFunctionId;
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  utils::Optional<am::plugin_manager::RPCPlugin> mock_rpc_plugin_opt(
+      mock_rpc_plugin_);
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(_, _))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+  ON_CALL(mock_rpc_plugin_, PluginName())
+      .WillByDefault(Return(application_manager::plugin_manager::plugin_names::
+                                app_service_rpc_plugin));
+  ON_CALL(mock_rpc_plugin_, GetCommandFactory())
+      .WillByDefault(ReturnRef(mock_command_factory_));
+  ON_CALL(mock_command_factory_, IsAbleToProcess(_, _))
+      .WillByDefault(Return(true));
+  EXPECT_TRUE(rpc_service_->IsAppServiceRPC(function_id, source));
+}
+
+TEST_F(RPCServiceImplTest,
+       UpdateMobileRPCParams_FunctionSchemaIsResettedAndAddedCustomMember) {
+  using namespace ns_smart_device_link::ns_smart_objects;
+  // Get initial state of schema
+  auto function_id = mobile_apis::FunctionID::OnHMIStatusID;
+  auto message_type = mobile_apis::messageType::notification;
+  MobileAPIDerived derived_api;
+  derived_api.SetSchema(function_id, message_type);
+  SetMobileAPI(derived_api);
+  CSmartSchema schema;
+  mobile_so_factory_.GetSchema(function_id, message_type, schema);
+  auto initial_schema_item = schema.getSchemaItem();
+  // Change state of schema and add custom parameter
+  auto new_function_id = mobile_apis::FunctionID::OnButtonEventID;
+  std::map<std::string, SMember> members;
+  members["custom member"] = SMember();
+  rpc_service_->UpdateMobileRPCParams(new_function_id, message_type, members);
+  mobile_so_factory_.GetSchema(new_function_id, message_type, schema);
```

Comment:
Create separate updated schema. Try not to mix different logical data in the save variable even if you can

Reply:
Fixed in commit c5c62b56e9a3bb72ab9810e870a69b0a17313311